pkg/controller/datadogagent/clusteragent.go

```diff
@@ -622,18 +622,25 @@ func (r *ReconcileDatadogAgent) manageClusterAgentRBACs(logger logr.Logger, dda
 
 	rbacResourcesName := getClusterAgentRbacResourcesName(dda)
 	clusterAgentVersion := getClusterAgentVersion(dda)
+
+	// Create ServiceAccount
+	serviceAccount := &corev1.ServiceAccount{}
+	if err := r.client.Get(context.TODO(), types.NamespacedName{Name: rbacResourcesName, Namespace: dda.Namespace}, serviceAccount); err != nil {
+		if errors.IsNotFound(err) {
+			return r.createServiceAccount(logger, dda, rbacResourcesName, clusterAgentVersion)
+		}
+		return reconcile.Result{}, err
+	}
 	// Create or update ClusterRole
 	clusterRole := &rbacv1.ClusterRole{}
 	if err := r.client.Get(context.TODO(), types.NamespacedName{Name: rbacResourcesName}, clusterRole); err != nil {
 		if errors.IsNotFound(err) {
 			return r.createClusterAgentClusterRole(logger, dda, rbacResourcesName, clusterAgentVersion)
 		}
 		return reconcile.Result{}, err
-	}
-	if result, err := r.updateIfNeededClusterAgentClusterRole(logger, dda, rbacResourcesName, clusterAgentVersion, clusterRole); err != nil {
+	} else if result, err := r.updateIfNeededClusterAgentClusterRole(logger, dda, rbacResourcesName, clusterAgentVersion, clusterRole); err != nil {
```

Comment:
Is this change functional or cosmetic here (joining the two checks into `else if`)? Personally I prefer to avoid else if when the conditions are complex like this and there are error checks involved - it makes it rather hard to follow and may make the code susceptible to misinterpretation and refactoring leading to bugs.

Reply:
agree, I have reverted this change